tests/nodes/NodeConnectionManager.termination.test.ts

```diff
@@ -408,6 +411,9 @@ describe(`${NodeConnectionManager.name} termination test`, () => {
       polykeyAgent = await PolykeyAgent.createPolykeyAgent({
         password,
         nodePath: nodePath,
+        networkConfig: {
+          proxyHost: '127.0.0.1' as Host,
```

Comment:
The `proxyHost` is the external facing host of the proxy connections now.

The default I believe is `::` or `0.0.0.0` which means bind to all interfaces. But during testing you are forcing it to be localhost to avoid polluting the network.

Is that your intention with this?

Reply:
I think in this case it's because later on the proxy host/port is retrieved using

```ts
await nodeGraph.setNode(agentNodeId, {
  host: polykeyAgent.proxy.getProxyHost(),
  port: polykeyAgent.proxy.getProxyPort(),
});
```

Which would return `0.0.0.0` (the default value) if not set explicitly to `127.0.0.1`. Alternatively instead of using `polykeyAgent.proxy.getProxyHost()` to get the host you would need to set it to `127.0.0.1` there instead, otherwise you'll be trying to connect to `0.0.0.0`.

Reply:
What @emmacasolin said was the idea. It's the easiest way to ensure that everything has the correct IP, including the logging.

Reply:
If `getProxyHost()` returns `0.0.0.0`. This is technically correct. Because it is infact hosted on all IP addresses in the OS.

This is why if you want to target a server, that may have been bound on `0.0.0.0`, you must then use `networkUtils.resolvesZeroIP()` to ensure that `0.0.0.0` or `::` gets "resolved" to `127.0.0.1` or `::1`. You cannot just use the host parameter as is.

I provisioned for this before inside `Connection` constructor which uses that `resolvesZeroIP`, but I didn't expect the confusion that would arise from logging, which would still use the original address of `0.0.0.0` or `::` when the `Connection` was constructed. This conversion still exists in `Connection` in this PR.

What we changed here, is to actually disallow targeting wildcard addresses when attempting to connect to another host in the whole `network/Proxy.ts`, even though internally it can handle it. I think this is good because it ensures explicit checking and prevents confusion with our logging.

In the future, we could use the type system to help prevent this. If we had "negation" types. That is a function could accept `Host` type EXCEPT where the `Host` is a wildcard address. https://catchts.com/type-negation

Reply:
Anything here I need to address in this PR? or is this some notes on what needs to be done at some point?

Reply:
Nothing to change. It's good.